crates/amaru-sim/src/echo/message.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
+pub struct Envelope {
+    pub id: u64,
```

Comment:
Is this needed, given that all messages already have a msg_id?

Reply:
Unfortunately yes, I am unsure why but this is part of the messages' schema 🤷 
```
INFO [2025-02-17 09:03:23,004] n1 stderr - maelstrom.process {"timestamp":"2025-02-17T08:03:23.004123Z","level":"INFO","fields":{"message":"input '{\"id\":76,\"src\":\"c2\",\"dest\":\"n1\",\"body\":{\"echo\":\"Please echo 107\",\"type\":\"echo\",\"msg_id\":38}}\n'"},"target":"amaru_sim::echo::run","span":{"name":"execute"},"spans":[{"stage":"read","name":"stage"},{"name":"execute"}]}
```

Reply:
Hmm, you're right... That's odd.

For what it's worth, the `id` field isn't listed in the protocol:  https://github.com/jepsen-io/maelstrom/blob/main/doc/protocol.md#messages .

I didn't even notice that it was there in my implementation (I assume the JSON deserialiser simply ignored it).

Reply:
I know, that's also why I said I found implementing the protocol a bit confusing

Reply:
Does serde complain that there are fields you didn't specify in the target datatype of the deserialisation, or how did you even notice this? Can you tell it to ignore fields you don't have in you Message datatype?